server/container/guice/cassandra-rabbitmq-guice/src/test/java/org/apache/james/modules/CassandraRabbitMQJMAPTestModule.java

```diff
@@ -0,0 +1,42 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.modules;
+
+import static org.apache.james.CassandraRabbitMQJamesServerMain.ALL_BUT_JMX_CASSANDRA_RABBITMQ_MODULE;
+
+import org.apache.james.GuiceJamesServer;
+import org.apache.james.JamesServerExtensionBuilder;
+import org.apache.james.mailbox.extractor.TextExtractor;
+import org.apache.james.mailbox.store.search.PDFTextExtractor;
+import org.apache.james.webadmin.WebAdminConfiguration;
+
+import com.google.inject.Module;
+import com.google.inject.util.Modules;
+
+public interface CassandraRabbitMQJMAPTestModule {
+    JamesServerExtensionBuilder.ServerProvider DEFAULT_CASSANDRA_JMAP_SERVER = configuration -> GuiceJamesServer.forConfiguration(configuration)
+        .combineWith(CassandraRabbitMQJMAPTestModule.DEFAULT);
+
+    Module DEFAULT = Modules.override(ALL_BUT_JMX_CASSANDRA_RABBITMQ_MODULE)
+        .with(TestJMAPServerModule.DEFAULT,
+            binder -> binder.bind(TextExtractor.class).to(PDFTextExtractor.class));
+
+    Module ENABLE_WEBADMIN = binder -> binder.bind(WebAdminConfiguration.class).toInstance(WebAdminConfiguration.TEST_CONFIGURATION);
```

Comment:
It is duplicated third times (one in memory, one in cassandra jmap). Any way to resolve that issue?

Reply:
Introduce this in Guice webadmin project. But it would lead anyway to several imports.